src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/inference/InsDelVariantDetector.java

```diff
@@ -164,9 +163,71 @@ public void inferSvAndWriteVCF(final JavaRDD<AssemblyContigWithFineTunedAlignmen
+        final StrandSwitch strandSwitch = novelAdjacencyReferenceLocations.strandSwitch;
+
+        final SimpleSVType type;
+        if (strandSwitch == StrandSwitch.NO_SWITCH) { // no strand switch happening, so no inversion
+            if (start==end) { // something is inserted
+                final boolean hasNoDupSeq = !novelAdjacencyReferenceLocations.complication.hasDuplicationAnnotation();
+                final boolean hasNoInsertedSeq = novelAdjacencyReferenceLocations.complication.getInsertedSequenceForwardStrandRep().isEmpty();
+                if (hasNoDupSeq) {
+                    if (hasNoInsertedSeq) {
+                        throw new GATKException("Something went wrong in type inference, there's suspected insertion happening but no inserted sequence could be inferred "
+                                + novelAdjacencyReferenceLocations.toString());
+                    } else {
+                        type = new SimpleSVType.Insertion(novelAdjacencyReferenceLocations); // simple insertion (no duplication)
+                    }
+                } else {
+                    if (hasNoInsertedSeq) {
+                        type = new SimpleSVType.DuplicationTandem(novelAdjacencyReferenceLocations); // clean expansion of repeat 1 -> 2, or complex expansion
+                    } else {
+                        type = new SimpleSVType.DuplicationTandem(novelAdjacencyReferenceLocations); // expansion of 1 repeat on ref to 2 repeats on alt with inserted sequence in between the 2 repeats
+                    }
+                }
+            } else {
+                final boolean hasNoDupSeq = !novelAdjacencyReferenceLocations.complication.hasDuplicationAnnotation();
+                final boolean hasNoInsertedSeq = novelAdjacencyReferenceLocations.complication.getInsertedSequenceForwardStrandRep().isEmpty();
+                if (hasNoDupSeq) {
+                    if (hasNoInsertedSeq) {
+                        type = new SimpleSVType.Deletion(novelAdjacencyReferenceLocations); // clean deletion
+                    } else {
+                        type = new SimpleSVType.Deletion(novelAdjacencyReferenceLocations); // scarred deletion
+                    }
+                } else {
+                    if (hasNoInsertedSeq) {
+                        type = new SimpleSVType.Deletion(novelAdjacencyReferenceLocations); // clean contraction of repeat 2 -> 1, or complex contraction
+                    } else {
+                        throw new GATKException("Something went wrong in type inference, there's suspected deletion happening but both inserted sequence and duplication exits (not supported yet): "
+                                + novelAdjacencyReferenceLocations.toString());
+                    }
+                }
+            }
+        } else {
+            type = new SimpleSVType.Inversion(novelAdjacencyReferenceLocations);
+        }
+
+        // developer check to make sure new types are treated correctly
```

Comment:
Can we remove this code block? I don't see how it can actually fail..

Reply:
removed